test/testdata/rewriter/prop_raise_on_nil_write.rb

```diff
@@ -0,0 +1,5 @@
+# typed: strict
+class Thing
+  include T::Props
+  prop :foo, T.nilable(String), raise_on_nil_write: true
+end
```

Comment:
Could you add some cases that show the behavior of writes with nil values?

Reply:
Added a few tests, both in the rewriter and also in the runtime for struct.